src/popup/pages/AppsBrowser.vue

```diff
@@ -155,43 +158,46 @@ export default defineComponent({
       () => !!modalsOpen.value.find((modal) => modal.name === MODAL_WARNING_DAPP_BROWSER),
     );
 
+    async function removeRpcClientIfAny() {
+      if (!currentClientId.value) return;
+      clearInterval(shareWalletInfoInterval);
+      const sdk = await getAeSdk();
+      sdk.removeRpcClient(currentClientId.value);
+      currentClientId.value = '';
+    }
+
     async function onAppLoaded() {
       if (!iframeRef.value || !selectedApp.value) return;
+      // Don't recreate RpcClient in Safari desktop and iOS webview
+      // because on these platforms `load` event triggers on anchor navigation
+      if (IS_SAFARI && currentClientId.value) return;
+      await removeRpcClientIfAny();
       const sdk = await getAeSdk();
-      let isAddedToHistory = false;
-
       const target = iframeRef.value.contentWindow;
-
-      try {
-        const connection = new BrowserWindowMessageConnection({
-          target,
-          origin: undefined,
-        });
-        currentClientId.value = sdk.addRpcClient(connection);
-
-        sdk.shareWalletInfo(currentClientId.value);
-
-        shareWalletInfoInterval = setInterval(
-          () => {
-            try {
-              const rpcClient = sdk._clients.get(currentClientId.value);
-              if (rpcClient) {
-                sdk.shareWalletInfo(currentClientId.value);
-
-                if (rpcClient.status === RPC_STATUS.CONNECTED && !isAddedToHistory) {
-                  isAddedToHistory = true;
-                  addHistoryItem(selectedApp.value);
-                }
-              }
-            } catch (e) {
-              handleUnknownError(e);
+      const connection = new BrowserWindowMessageConnection({ target });
+      currentClientId.value = sdk.addRpcClient(connection);
+      shareWalletInfoInterval = executeAndSetInterval(
+        () => {
+          const rpcClient = sdk._getClient(currentClientId.value);
+          if (
+            rpcClient.status === RPC_STATUS.CONNECTED
+            && lastUrlAddedToHistory !== selectedApp.value.url
+          ) {
+            lastUrlAddedToHistory = selectedApp.value.url;
+            addHistoryItem(selectedApp.value);
+          }
```

Comment:
This code don't need to wrap with try/catch because it is all local, here is nothing to fail